Source/events.handling/Builder/EventHandlerClassBuilder.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Logger } from 'winston';
+
+import { Guid } from '@dolittle/rudiments';
+import { Constructor } from '@dolittle/types';
+
+import { IContainer } from '@dolittle/sdk.common';
+import { EventTypeMap } from '@dolittle/sdk.events';
+import { ExecutionContext } from '@dolittle/sdk.execution';
+import { Cancellation } from '@dolittle/sdk.resilience';
+
+import { EventHandlersClient } from '@dolittle/runtime.contracts/Runtime/Events.Processing/EventHandlers_grpc_pb';
+
+import { EventHandler, EventHandlerSignature, IEventHandlers, internal } from '../index';
+import { EventHandlerDecoratedTypes } from './EventHandlerDecoratedTypes';
+import { eventHandler as eventHandlerDecorator } from './eventHandlerDecorator';
+import { handles as handlesDecorator } from './handlesDecorator';
+import { HandlesDecoratedMethods } from './HandlesDecoratedMethods';
+import { EventType, EventTypeId, Generation, IEventTypes } from '@dolittle/sdk.artifacts';
+import { HandlesDecoratedMethod } from './HandlesDecoratedMethod';
+import { ICanBuildAndRegisterAnEventHandler } from './ICanBuildAndRegisterAnEventHandler';
+
+export class EventHandlerClassBuilder<T> implements ICanBuildAndRegisterAnEventHandler {
+    constructor(private readonly _eventHandlerType: Constructor<T>, private readonly _instance?: T) {
+    }
+
+    /** @inheritdoc */
+    buildAndRegister(
+        client: EventHandlersClient,
+        eventHandlers: IEventHandlers,
+        container: IContainer,
+        executionContext: ExecutionContext,
+        eventTypes: IEventTypes,
+        logger: Logger,
+        cancellation: Cancellation): void {
+
+        logger.debug(`Building event handler of type ${this._eventHandlerType.name}`);
+        const decoratedType = EventHandlerDecoratedTypes.types.find(_ => _.type === this._eventHandlerType);
+        if (decoratedType == null) {
+            logger.warning(`The event handler class ${this._eventHandlerType.name} needs to be decorated with an @${eventHandlerDecorator.name} decorator`);
+            return;
+        }
+        logger.debug(`Building ${decoratedType.partitioned ? 'partitioned' : 'unpartitioned'} event handler ${decoratedType.eventHandlerId} processing events in scope ${decoratedType.scopeId} from type ${this._eventHandlerType.name}`);
+        const methods = HandlesDecoratedMethods.methodsPerEventHandler.get(this._eventHandlerType);
+        if (methods == null) {
+            logger.warning(`There are no event handler methods to register in event handler ${this._eventHandlerType.name}. An event handler method needs to be decorated with @${handlesDecorator.name} or have the name {MethodName}`);
```

Comment:
A little C# that has snuck through? 😛 
```suggestion
            logger.warning(`There are no event handler methods to register in event handler ${this._eventHandlerType.name}. An event handler method must be decorated with @${handlesDecorator.name}`);
```